### PR TITLE
4762 improve documentation for vendors

### PR DESCRIPTION
**Pull request by SamJamCul**
## Context
https://trello.com/c/Oxa0e6yD/4762-improve-documentation-for-vendors

## Changes proposed in this pull request
Updated the draft api docs to cover some more details. This includes:

 - Adding in missing field lengths
 - Adding state transition links to descriptions for actions that can result in a state transition
 - Adding to the v1.1 proposed changes

Currently in prod our longest ProviderUser first name and last name would total less than 50 chars, so a limit of 100 seems generous enough!

image

## Guidance to review
Head over to /api-docs/draft to peruse the changes!

### CODE REVIEW

File: app/views/api_docs/vendor_api_docs/reference/draft.html.erb [hunk 1]
```diff
@@ -128,6 +132,29 @@ We will also support requesting a specific minor version via the URL. For exampl
   If an application is withdrawn at a candidate's request the field <code>withdrawn_or_declined_for_candidate</code> will be set to true to determine the difference between a normal withdrawal and one at a candidate's request. Also either the <code>offer_declined_at</code> field or the <code>date</code> field, inside the existing <%= govuk_link_to 'Withdrawal Object', '#withdrawal-object' %>, will be populated with the timestamp the application was withdrawn or declined.
 </p>
 
+<p class="govuk-body">Only applications in the <code>offer</code> state will transition to the <code>declined</code> state.</p>
+
+<p class="govuk-body">Applications in the following states will transition to the <code>withdrawn</code> state:</p>
+<ul class="govuk-list">
```
Comment (duncanjbrown) [inline app/views/api_docs/vendor_api_docs/reference/draft.html.erb:138]: This (and the below) can go out of date — could we generate this from the state machine?

File: config/vendor_api/draft.yml [hunk 2]
```diff
@@ -598,6 +614,7 @@ components:
           enum:
           - pending_conditions
           - recruited
+          maxLength: 18
```
Comment (cwrw) [inline config/vendor_api/draft.yml:617]: could we go to 30 to give us some room incase we add a new status?

File: config/vendor_api/draft.yml [hunk 3]
```diff
@@ -560,6 +574,7 @@ components:
           description: |
             The unique ID of this note. Automatically generated by our service.
           example: 123456
+          maxLength: 18
```
Comment (cwrw) [inline config/vendor_api/draft.yml:577]: could we update the interview ID as well to 18?

File: app/views/api_docs/vendor_api_docs/reference/draft.html.erb [hunk 4]
```diff
@@ -102,13 +102,17 @@ We will also support requesting a specific minor version via the URL. For exampl
 
 <p class="govuk-body">If an applicant has an interview, the <code>interviews</code> array in the application response will be populated and can be used to determine the presence of any interviews, including cancelled ones. The status of the application will remain as <code>awaiting_provider_decision</code>.</p>
 
+<p class="govuk-body">When an applicant accepts an offer, all upcoming interviews on all of their applications are automatically cancelled.</p>
```
Comment (cwrw) [inline app/views/api_docs/vendor_api_docs/reference/draft.html.erb:105]: could we also add a note around if an application is withdrawn/declined as well

File: config/vendor_api/draft.yml [hunk 5]
```diff
@@ -192,6 +195,8 @@ paths:
       summary: Create interview
       description: |
         Creates a new interview for this application.
+
+        This will transition the application to the [interviewing state](/api-docs/lifecycle#interviewing) if it is not already in this state.
```
Comment (cwrw) [inline config/vendor_api/draft.yml:199]: we don't surface interviewing state in the API, so might be confusing, I suggest we leave it out

**Review by cwrw — COMMENTED** [review]
looking good, some minor comments

**Review by despo — APPROVED** [review]

**Pull request merged**